hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java

```diff
@@ -245,16 +246,18 @@ public static boolean isConsistentLogicalTimestampEnabled(Configuration conf) {
    */
   public static boolean needsGuardByLock(Configuration conf) {
     return conf.getBoolean(FlinkOptions.METADATA_ENABLED)
-        || conf.getString(HoodieWriteConfig.WRITE_CONCURRENCY_MODE.key(), HoodieWriteConfig.WRITE_CONCURRENCY_MODE.defaultValue())
-            .equalsIgnoreCase(WriteConcurrencyMode.OPTIMISTIC_CONCURRENCY_CONTROL.value());
+        || ConfigUtils.resolveEnum(WriteConcurrencyMode.class, conf.getString(
```

Comment:
nit: we can just convert to upper case and avoid using `ConfigUtils.resolveEnum` here?